lib/appsignal/garbage_collection_profiler.rb

```diff
@@ -0,0 +1,29 @@
+module Appsignal
+  class GarbageCollectionProfiler
+    def initialize
+      @total_time = 0
+    end
+
+    def total_time
+      lock.synchronize do
+        @total_time += (internal_profiler.total_time * 1000).round
+        internal_profiler.clear
+      end
+      @total_time
+    end
+
+    private
+
+    def self.lock
+      @lock ||= Mutex.new
+    end
+
+    def internal_profiler
+      GC::Profiler
+    end
+
+    def lock
+      self.class.lock
+    end
+  end
+end
```

Comment:
Why expose the lock if we only use it internally? You could also only put `@lock = Mutex.new` in the initializer.